content/2022-04-14-our-secure-supply-chain-obsession.md

```diff
@@ -0,0 +1,384 @@
+where SBOMs come into play.
+
+If no one can ever truly guarantee their software is 100% invulnerable in
+perpetuity, the best and most honest gesture anyone can make to gain the trust
+of their users is to be publicly transparent about what's in the software. This
+can be instrumental in helping future users assess what new vulnerabilities
+impacting your software have been discovered in the time since you released it.
+
+As it turns out, the fine people behind Grype also make a tool called
+[Syft](https://github.com/anchore/syft) that can scan a Docker image to generate
+an SBOM in a variety of formats. (For what it's worth, Grype seems to actually
+_use_ Syft under the covers.)
+[SPDX](https://en.wikipedia.org/wiki/Software_Package_Data_Exchange)
+(Software Package Data Exchange) is the _de facto_ standard for expressing SBOMs
+-- and this itself can be represented in a number of different formats, with
+JSON seemingly being the most popular.
+
+It takes little more than a command like this to generate an SBOM for a Docker
+image:
+
+```shell
+$ syft -o spdx-json debian:bullseye
+```
+
+In recent weeks, the Brigade project has integrated this into our release
+processes. Just as our CI processes automatically scan images with Grype
+immediately after they're built, our release processes now automatically
+generate SBOMs with Syft and publish them immediately after pushing images to
+to Docker Hub.
+
+This is where we'll freely admit that we're pretty sure there's still room to
+improve upon what we're doing with these SBOMs. For the moment, we're publishing
+the SBOM for every image we push to its corresponding GitHub release page. This
+seems, to us, to be better than nothing, but we expect to discover more useful
+things to do with these over time. Although I can claim no expertise on this
+topic, I do possess a vague notion that OCI registries (Docker image registries)
+are rapidly improving in terms of their ability to attach arbitrary metadata to
+images. Some vendors may be ahead of others. I expect, eventually, we'll be able
+to _seamlessly_ push our SBOMs to OCI registries with the images themselves.
+
+Recalling once again that the cost to secure the software supply chain is
+eternal vigilance, Brigade users can be assured that maintainers will continue
+to evolve our process as the state of the art advances.
+
+## Signing Images
+
+Last, but certainly not least, in recent weeks the project has also begun
+signing all images we push to Docker Hub. While signing Docker images is not a
+new or novel concept, it's something we hadn't been doing previously, and in
+light of our recent efforts to minimize our dependencies, scan our images, and
+publish bills of material, we wanted to go just one step further to help our
+users know that the images they're pulling from Docker Hub are in fact the
+_authentic_ images to which this great diligence has been applied.
```

Comment:
You went into a lot of detail about scanning and generating sboms, but you didn't really discuss signing here.  It might be helpful to include some details about that and how people can verify the images that you are producing.

Reply:
I felt like the process was relatively complicated and documented better elsewhere. I was previously linking to Docker's own docs about this and that fell out in my last re-write of this section. Maybe just put that back?

Reply:
I put it back in c4ea9e90bfd6869adb889c8677ed4b03f6487b18